01-JS-DOM-APIs/javascript.js

```diff
@@ -0,0 +1,117 @@
+//1-4
+function show(){
+                  let obj = document.getElementById("hidden");
+                  if(obj.style.display == "block"){
+                        obj.style.display = "none";
+                        document.getElementById("buttonShow").value= "Show";
+                      }
+                  else{
+                        obj.style.display = "block";
+                        document.getElementById("buttonShow").value= "Hide";
+                      }
+                    }
+//5-6
+function buttonJokeWithoutPromise() {
+    let xhttp = new XMLHttpRequest();
+    xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+    xhttp.onreadystatechange = function(event) {
+    if (xhttp.readyState === XMLHttpRequest.DONE)
+    {
+      let response = JSON.parse(event.target.response);
+      let txt = response.value.joke;
+      document.getElementById("jokeHidden").innerHTML = response.value.joke;
+      document.getElementById("buttonjoke").value="Another joke";
+               }
+           }
+    xhttp.send();
+      }
+//7-9
+function buttonJokeWithPromise(){
+            let config = {
+              url: "http://api.icndb.com/jokes/random"
+                  };
+                        let xhttp = new XMLHttpRequest();
+                        function resolve(xhttp) {
+                            let response = JSON.parse(event.target.response);
+                             document.getElementById("jokeHidden2").innerHTML = response.value.joke;
+                             if (xhttp.readyState == XMLHttpRequest.DONE) {
+                               if (xhttp.responseText.value) {
+                                 document.getElementById("button2").innerHTML  = xhttp.responseText.value.joke;
+                               }
+                             }
+                         }
+                         function reject() {
+                           document.getElementById("all").style.backgroundColor = "red";
+                         }
+                         eventRequestReusable(config).then(resolve,reject);
+                        };
+
+function eventRequestReusable(object, resolve, reject) {
```

Comment:
This looks good, but you can name it as something more descriptive as *makeAjaxCall* or *performRequest* or something like that, to make really easy at first glance to know what this function does, also try to avoid generic parameters or variable names as *object*, that object can be named *requestConfig* or something like that, that way when someone wants to use this function it will know right away what parameters this function expects.
Last detail regarding this, the *GET* method shouldn't be hardcoded, and instead should be part of the config parameter, as it is this function only allows to make get requests, what if in the future you need it for posts calls?